Samples-Typescript/DashboardLayout/dashboardLayout.ts

```diff
@@ -0,0 +1,155 @@
+      console.log('Initializing extension API');
+      await tableau.extensions.initializeAsync();
+
+      const dashboard = tableau.extensions.dashboardContent.dashboard;
+      this.dashboardObjects = dashboard.objects;
+      console.log(this.dashboardObjects);
+
+      // enabling dashboard event button
+      this._$('#dashboard-event-btn').prop('disabled', false);
+      this._$('#dashboard-event-btn').click(this.onEventButtonClick.bind(this));
+    }
+
+    // When changes are made to the dashboard we get all the details for each of the
+    // dashboard objects that were changed and compare it with their previous values.
+    // The dashboardLayoutChangeDetails property is a map of dashboard obj3ct ids to
+    // an array of dashboard layout changes.
+    // Dashboard layout change events are invoked when dashboard objects are resized, repositioned, added, removed,
+    // hidden, shown, renamed, made floating, and made tiled. Extension reloads when worksheets are added / removed.
+    private onDashboardLayoutChange(event: TableauEvent) {
+      console.log(event);
+      const dashboardEvent = event as DashboardLayoutChangedEvent;
+      const dashboardEventDetails = dashboardEvent.dashboardLayoutChangeDetails;
+      const dashboard = tableau.extensions.dashboardContent.dashboard;
+
+      // updating dashboard objects and storing the previous dashboard objects for referrence.
+      const oldDashboardObjects = this.dashboardObjects;
+      this.dashboardObjects = dashboard.objects;
+
+      // An empty dashboard layout change event may be invoked when loading an extension from the manifest.
+      // In this case we ignore it and return.
+      if (dashboardEventDetails === undefined || dashboardEventDetails.size === 0) {
+        return;
+      }
+
+      // Emptying previous content from the UI's change list.
+      this._$('#dashboard-layout-change-list').empty();
+
+      // Updating UI's change list to display information on the current dashboard event.
+      dashboardEventDetails.forEach((changesMade: DashboardLayoutChange[], dashboardObjectId: number) => {
+        // getting dashboard object from its id
+        const dashboardObject = dashboard.getDashboardObjectById(dashboardObjectId);
+
+        // building a div for the changes made to this dashboard object.
+        const changesDiv = this._$('<div>');
+
+        // checking if this dashboard object was added as part of the event.
+        if (changesMade.includes(tableau.DashboardLayoutChange.Added)) {
+          const toAppend = this._$('<h6/>');
+          toAppend.text(`Dashboard Object ${dashboardObjectId} added: "${dashboardObject.name}"`);
+          changesDiv.append(toAppend);
+          this._$('#dashboard-layout-change-list').append(changesDiv);
+          return;
+        }
+
+        // getting old dashboard object before event to compare it with the current one.
+        let oldDashboardObject;
+        // tslint:disable-next-line: prefer-for-of
```

Comment:
change this block of code to something like
let oldDashboardObject = oldDashboardObjects.find(o => o.id === dashboardObjectId)

and remove the tslint ignore